src/renderer/renderer.js

```diff
@@ -396,11 +663,35 @@ export class SDFGRenderer {
         this.canvas_manager.draw_async();
     }
 
+    // --------------------------------------------------------------
+    // These functions are only implemented in the context of VSCode.
+
+    send_new_sdfg_to_vscode() {}
+
+    add_node_to_graph() {}
+
+    remove_graph_nodes() {}
+
+    update_new_element() {}
+
+    show_no_daemon_dialog() {}
+
+    show_select_library_node_dialog() {}
```

Comment:
why are they here then?

Reply:
My thought was that it would be better to avoid an undefined reference if they are ever mistakingly called, and instead just do nothing. These functions are called from within the base renderer, theoretically it should just never get there outside the context of VSCode. This is, however, a debatable assumption. What do you think?

Reply:
I wouldn't say it's great design-wise but given our past issues I think that's fine for now.